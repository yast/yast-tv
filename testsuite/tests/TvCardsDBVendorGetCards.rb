# encoding: utf-8

module Yast
  class TvCardsDBVendorGetCardsClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      TEST_It_Now(nil, nil)

      TEST_It_Now(
        {
          "module"     => "bttv",
          "name"       => "my autodetected card",
          "unique_key" => "123xyz"
        },
        nil
      )

      # autodetected card has index 0
      TEST_It_Now({ "name" => "my autodetected card" }, 0)

      TEST_It_Now({ "name" => "my autodetected card" }, 1)

      TEST_It_Now(nil, 1)

      nil
    end

    def TEST_It_Now(auto, vendor)
      auto = deep_copy(auto)
      DUMP("---------------")

      # cards_database cannot be nil (see ReadCardsDatabase)
      Tv.cards_database = nil
      items = Tv.CardsDBVendorGetCards(auto, vendor)
      DUMP(items)

      Tv.cards_database = []
      items = Tv.CardsDBVendorGetCards(auto, vendor)
      DUMP(items)

      Tv.cards_database = [
        {
          "cards" => [
            {
              "module"     => "bttv",
              "name"       => "Unknown bttv card",
              "parameters" => { "card" => "0" }
            },
            {
              "module"     => "bttv",
              "name"       => "TView99 CPH06X",
              "parameters" => { "card" => "38" }
            }
          ],
          "name"  => "Other vendors"
        },
        {
          "cards" => [
            {
              "module"     => "bttv",
              "name"       => "ATI TV-Wonder VE",
              "parameters" => { "card" => "64" }
            }
          ],
          "name"  => "ATI"
        }
      ]
      items = Tv.CardsDBVendorGetCards(auto, vendor)
      DUMP(items)

      nil
    end
  end
end

Yast::TvCardsDBVendorGetCardsClient.new.main
