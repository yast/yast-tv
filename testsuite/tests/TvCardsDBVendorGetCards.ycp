{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

	define TEST_It_Now(map auto, integer vendor) ``{
	DUMP("---------------");

	// cards_database cannot be nil (see ReadCardsDatabase)
	Tv::cards_database = nil;
	list items = Tv::CardsDBVendorGetCards(auto,vendor);
	DUMP(items);

	Tv::cards_database = [];
	items = Tv::CardsDBVendorGetCards(auto,vendor);
	DUMP(items);

	Tv::cards_database = [
	$["cards": [
		$["module":"bttv", "name":"Unknown bttv card", "parameters":$["card":"0"]],
		$["module":"bttv", "name":"TView99 CPH06X", "parameters":$["card":"38"]]],
	  "name":"Other vendors"],
	$["cards": [
		$["module":"bttv", "name":"ATI TV-Wonder VE", "parameters":$["card":"64"]]],
	  "name":"ATI"]
	];
	items = Tv::CardsDBVendorGetCards(auto, vendor);
	DUMP(items);

	}

    TEST_It_Now(nil, nil);

    TEST_It_Now( $["module":"bttv",
				   "name":"my autodetected card",
				   "unique_key":"123xyz"
				 ], nil);

	// autodetected card has index 0
	TEST_It_Now( $["name":"my autodetected card" ], 0);

	TEST_It_Now( $["name":"my autodetected card" ], 1);

	TEST_It_Now( nil, 1);
}
