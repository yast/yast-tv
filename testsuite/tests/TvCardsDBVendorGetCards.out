Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.InstMode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	---------------
Dump	[]
Dump	[]
Dump	[]
Dump	---------------
Dump	[]
Dump	[]
Dump	[]
Dump	---------------
Dump	[$["name":"my autodetected card"]]
Dump	[$["name":"my autodetected card"]]
Dump	[$["name":"my autodetected card"]]
Dump	---------------
Dump	[]
Dump	[]
Dump	[$["module":"bttv", "name":"Unknown bttv card", "parameters":$["card":"0"]], $["module":"bttv", "name":"TView99 CPH06X", "parameters":$["card":"38"]]]
Dump	---------------
Dump	[]
Dump	[]
Dump	[$["module":"bttv", "name":"ATI TV-Wonder VE", "parameters":$["card":"64"]]]
